[MGR] Optimizing the equations and wording in Latent Discriminant Ananlysis User Guide

LiutongZhou opened the pull request:
<!--
Thanks for contributing a pull request! Please ensure you have taken a look at
the contribution guidelines: https://github.com/scikit-learn/scikit-learn/blob/master/CONTRIBUTING.md#pull-request-checklist
-->


<!--
Example: Fixes #1234. See also #3456.
Please use keywords (e.g., Fixes) to create link to the issues or pull requests
you resolved, so that they will automatically be closed when your pull request
is merged. See https://github.com/blog/1506-closing-issues-via-pull-requests
-->


# What does this implement/fix? Explain your changes.
1. Fix two typos in the documentation for [Linear Discriminant Analysis](http://scikit-learn.org/stable/modules/lda_qda.html#lda-qda).
2. Explicitly distinguish between data matrix X with n rows d columns and d dimensional random vector \vec{x}. The notations have been changed and __explicitly stated__:    
  * big X denotes n by d data, 
  * \vec{x} denotes d dimensional random vector.
3. Make notations consistent. 
4. Change the equation in section 1.2.3 to be consistent with the notation: data X is the n by d design matrix. 
5. Optimize wordings so they are mathematically rigorous.

The collected data X has d number of columns (d number of features). So this is d-dimensional Multivariate Gaussian.

In versions before 0.19.1, the PDF for d-dimensional Multivariate Gaussian Distribution is accidentally typed as 
![previous](https://latex.codecogs.com/gif.latex?p%28X%20%7C%20y%3Dk%29%20%3D%20%5Cfrac%7B1%7D%7B%282%5Cpi%29%5En%20%7C%5CSigma_k%7C%5E%7B1/2%7D%7D%5Cexp%5Cleft%28-%5Cfrac%7B1%7D%7B2%7D%20%28X-%5Cmu_k%29%5Et%20%5CSigma_k%5E%7B-1%7D%20%28X-%5Cmu_k%29%5Cright%29). 

n should be d/2. 

Besides, readers may feel confused if they think X is the design matrix. In fact, __X in the above equation is the random vector \vec{x}__

I correct it to be 
![now](https://latex.codecogs.com/gif.latex?p%28%5Cvec%7Bx%7D%20%7C%20y%3Dk%29%20%3D%20%5Cfrac%7B1%7D%7B%282%5Cpi%29%5E%7Bd/2%7D%20%7C%5CSigma_k%7C%5E%7B1/2%7D%7D%5Cexp%5Cleft%28-%5Cfrac%7B1%7D%7B2%7D%20%28%5Cvec%7Bx%7D-%5Cvec%7B%5Cmu%7D_k%29%5ET%20%5CSigma_k%5E%7B-1%7D%20%28%5Cvec%7Bx%7D-%5Cvec%7B%5Cmu%7D_k%29%5Cright%29)


My revision is verified against [Wiki](https://en.wikipedia.org/wiki/Multivariate_normal_distribution)



<!--
Please be aware that we are a loose team of volunteers so patience is
necessary; assistance handling other issues is very welcome. We value
all user contributions, no matter how minor they are. If we are slow to
review, either the pull request needs some benchmarking, tinkering,
convincing, etc. or more likely the reviewers are simply busy. In either
case, we ask for your understanding during the review process.
For more information, see our FAQ on this topic:
http://scikit-learn.org/dev/faq.html#why-is-my-pull-request-not-getting-any-attention.

Thanks for contributing!
-->

LiutongZhou committed:
Correct the PDF for Multivariate Gaussian

This is a typo in the user manual.

The collected data X has d number of columns, n number of rows. So this is d-dimensional Multivariate Gaussian.

agramfort commented:
can you also add to the doc what d is ?

then +1 for merge

LiutongZhou committed:
Correct a typo in LDA equation 

Optimized the mathematical wording of LDA and equation for pdf of Multivariate Gaussian

lesteve commented:
For easier comparison:

This PR:
https://21005-843222-gh.circle-artifacts.com/0/doc/modules/lda_qda.html#mathematical-formulation-of-the-lda-and-qda-classifiers

dev doc:
http://scikit-learn.org/dev/modules/lda_qda.html#mathematical-formulation-of-the-lda-and-qda-classifiers

General advice: try to change the minimal amount of stuff, for example you added arrows on top of mu_k but then it is inconsistent with the equations afterwards so I would revert that.

Also it seems like the `d` looks a bit weird (i.e. not like the `n`) in "n is the number of observations and d is the number of features".

LiutongZhou committed:
Make notations in equations consistent

1. Rewrite equations to explicitly distinguish data X from random vector \vec{x}. 
2. Make notations consistent so people do not feel confused

LiutongZhou committed:
optimize the wording

rewording to make it more clear

LiutongZhou committed:
Optimize format for better rendering

LiutongZhou commented:
# For comparison

This Pull Request's Doc Build 21052 [https://21052-843222-gh.circle-artifacts.com/0/doc/modules/lda_qda.html#mathematical-formulation-of-the-lda-and-qda-classifiers](url)

dev doc:
http://scikit-learn.org/dev/modules/lda_qda.html#mathematical-formulation-of-the-lda-and-qda-classifiers

LiutongZhou committed:
Correct a typo in equation of section 1.2.3

lesteve commented:
> General advice: try to change the minimal amount of stuff, for example you added arrows on top of mu_k but then it is inconsistent with the equations afterwards so I would revert that. Same thing for the x.

You may have misunderstood my earlier comment: I asked you to change the minimal amount of things and you added yet more changes in your latest commits. Please understand that short focussed PRs are a lot more likely to get merged quickly.

It would be great if you could keep in this PR only these changes:
* replacing `n` by `d/2`
* explaining in a few words what `d` is (number of features)
* Not sure which typos you fixed, but typo fixes are fine too

LiutongZhou commented:
Why we need to distinguish data from random vector?

My explanation: 

Data X are observed __fixed__ values generated by (sampled from) the distribution of __random__ vector x.

jnothman commented:
In statistics talking about a random vector almost makes sense. But rather, statisticians choose to assume that a vector is a draw from a random variable of a particular distribution. The vector is not itself inherently random and calling it so is unhelpful to machine learning people who may be less familiar with that statistical mindset.

lesteve commented:
@LiutongZhou please follow what I said in https://github.com/scikit-learn/scikit-learn/pull/10902#issuecomment-378655639. This will ensure that this PR does not die with unnecessary and never-ending side-discussions.

In scikit-learn we have ~1000 opened issues and ~600 opened PRs, you need to be aware that arguing over minor things like this is not a efficient use of anybody's time and energy.

LiutongZhou committed:
Revert Back

1. Reverted back to commit 8e6d5603ff375eb399d38caf3f53713b5934a049
2. Added "where X is a d-dimensional random vector"

lesteve committed:
Tweaks

lesteve commented:
Merging this one, thanks @LiutongZhou!
